File: client/src/lib/invoke/checkInvoiceValues.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { Invoice } from "@bindings";

export function checkInvoiceValues(invoice: Invoice): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("check_invoice_values", { invoice }),
        (error) => `Failed to check invoice values: ${error}`
    );
}

File: client/src-tauri/src/commands/check_invoice_values.rs
use crate::core::invoice::Invoice;

#[tauri::command]
pub async fn check_invoice_values(invoice: Invoice) -> Result<(), String> {
    tauri::async_runtime::spawn_blocking(move || invoice.check_values())
        .await
        .map_err(|err| format!("Failed to check invoice values: {}", err))?
}
